### ISSUE TITLE
Avoid using `compile` configuration to declare dependencies in gradle build scripts

### ISSUE THREAD

**Issue by eparovyshnaya** [exercism/kotlin#368]
Currently Gradle reports a warning on the topic: `compile` configuration has been marked as deprecated for further removal. 

Suppose we should substitute it with `implementation`.

>The compile configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 7.0. Please use the implementation configuration instead.

**Comment by dector** [comment]
Absolutely! `compile()` is still used in build-configuration because it's required for current implementation of `compileStarterSourceKotlin` task (used in unit tests). This dependency should be reimplemented or fixed before using `implementation()`.
